scripts/cpm_env_setup.sh

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+
+TIMESTAMP=$(date +"%Y%m%d_%H%M%S")
+LOG_FILE="${TIMESTAMP}_cmp_env_setup.log"
+KEEP_PROGRESS_FILE=false
+
+if [[ "$1" == "--keep-progress-file" ]]; then
```

Comment:
About the `--keep-progress-file` switch: I didn't ask about it as it was quite straightforward what you meant but I wonder if it was affected by our `log file` vs `progress file` confusion.

Log file (`${TIMESTAMP}_cmp_env_setup.log`) is always kept and has all logging from the script and scripts it's calling.

Progress file (`$TOP/.onboarding_env_setup_progress`) is only used to keep track of which stages were completed, I don't think it's very useful to keep it. In addition, it sort of creates a bug, when after script successfully completes once, you cannot re-run it, as it will find progress file indicating all stages were completed.

Reply:
we discussed this, i think we agree on this